easybuild/easyblocks/c/clang.py

```diff
@@ -407,21 +408,54 @@ def build_with_prev_stage(self, prev_obj, next_obj):
         mkdir(next_obj)
         change_dir(next_obj)
 
-        # Configure.
-        CC = os.path.join(prev_obj, 'bin', 'clang')
-        CXX = os.path.join(prev_obj, 'bin', 'clang++')
+        # Make sure the clang and clang++ compilers from previous stages are (temporarily) in PATH
+        orig_env = copy.deepcopy(os.environ)
+        prev_obj_path = os.path.join(prev_obj, 'bin')
+        current_path = read_environment({'path': 'PATH'})
+        setvar('PATH', current_path['path'] + ":" + prev_obj_path)
+
+        # If building with rpath, create RPATH wrappers for the Clang compilers for stage 2 and 3
+        if build_option('rpath'):
+            from easybuild.toolchains.compiler.clang import Clang
+            my_clang_toolchain = Clang(name='Clang', version='1')
+            my_clang_toolchain.prepare_rpath_wrappers()
+            self.log.info("Prepared clang rpath wrappers")
+
+            # RPATH wrappers add -Wl,rpath arguments to all command lines, including when it is just compiling
+            # Clang by default warns about that, and then some configure tests use -Werror which turns those warnings
+            # into errors. As a result, those configure tests fail, even though the compiler supports the requested
+            # functionality (e.g. the test that checks if -fPIC is supported would fail, and it compiles without
+            # resulting in relocation errors).
+            # See https://github.com/easybuilders/easybuild-easyblocks/pull/2799#issuecomment-1270621100
+            # Here, we add -Wno-unused-command-line-argument to CXXFLAGS to avoid these warnings alltogether
+            cflags = os.getenv('CFLAGS')
+            cxxflags = os.getenv('CXXFLAGS')
+            setvar('CFLAGS', "%s %s" % (cflags, '-Wno-unused-command-line-argument'))
+            setvar('CXXFLAGS', "%s %s" % (cxxflags, '-Wno-unused-command-line-argument'))
 
+        # Configure.
         options = "-DCMAKE_INSTALL_PREFIX=%s " % self.installdir
```

Comment:
We should make `options` a list, that'll be a lot cleaner...

Reply:
Done by you in https://github.com/easybuilders/easybuild-easyblocks/pull/2799/commits/21442f8e8dcac3425bb0af18d623a49b5163df18